### ISSUE TITLE
Bug: Searching for Attributes Broken After Upgrading to 2.4.199

### ISSUE THREAD

**Issue by chennin7** [MISP/MISP#10007]
### Actual behavior

After we upgraded MISP to 2.4.199 from 2.4.194, when running a search on attributes through the MISP UI, we receive an internal server error for every search.

### Expected behavior

We should be able to use the search attributes function in MISP.

### Steps to reproduce

Run an attribute search on MISP 2.4.199. Note: this search works on our previous version, 2.4.194

Reproduction:
<img width="1287" alt="Screenshot 2024-10-24 at 4 00 11 PM" src="https://github.com/user-attachments/assets/c4a8e3e7-73fb-430f-857a-412fe7998335">
<img width="1287" alt="Screenshot 2024-10-24 at 3 59 21 PM" src="https://github.com/user-attachments/assets/e64c8785-5d08-4ed2-9bcc-d77a96ceb462">

### Version

2.4.199

### Operating System

Red Hat

### Operating System version

7.9

### PHP version

7.4.33

### Browser

Chrome

### Browser version

129.0.6668.103

### Relevant log output

```shell
Logs from error.log:

2024-10-24 15:10:45 Error: [Error] Call to a member function query() on null
Request URL: /attributes/search/results
Stack Trace:
#0 [internal function]: AttributesController->search()
#1 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(499): ReflectionMethod->invokeArgs()
#2 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(193): Controller->invokeAction()
#3 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke()
#4 /var/www/MISP/app/webroot/index.php(105): Dispatcher->dispatch()
#5 {main}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by tomking2** [comment]
This is due to https://github.com/MISP/MISP/blob/2.4/app/Controller/AttributesController.php#L1563

`MispAttribute` is a `2.5` specific model, whereas it's called `Attribute` in `2.4`. Looks like some 2.5 specific code found its way into 2.4. @iglocska more fodder for renaming the models in 2.4 :).

Happy to fix and send a PR, but I'm not sure how we should approach this without it finding it's way into 2.5 branch on a periodic sync of the codebases?

**Comment by iglocska** [comment]
Yep, indeed nice catch. Fixed for now!

**Comment by iglocska** [comment]
but yeah, should definitely backport the renaming to 2.4

**Comment by chennin7** [comment]
thanks y'all! I assume this fix will come in the 2.4.200 release?
